xarray/core/dataset.py

```diff
@@ -136,6 +136,14 @@ def merge_indexes(
         names, labels, levels = [], [], []
         current_index_variable = variables.get(dim)
 
+        for n in var_names:
+            var = variables[n]
+            if ((current_index_variable is not None) and
```

Comment:
nice: `is not` and `!=` have higher precedence than `and`, so you could remove these nested parantheses